overlays: Add ssd1306-spi overlay

MikeDK opened the pull request:
Add an overlay for SSD1306 based OLED boards using SPI.
This will load the staging fbtft driver.

Signed-off-by: Michael Kaplan <m.kaplan@evva.com>

MikeDK committed:
overlays: Add ssd1306-spi overlay

Add an overlay for SSD1306 based OLED boards using SPI.
This will load the staging fbtft driver.

Signed-off-by: Michael Kaplan <m.kaplan@evva.com>

MikeDK commented:
I just realized I forgot to provide overrides for width and height (as there are also 128x32 displays around on aliexpress etc)
But I am not sure how this looks like for a vendor parameter ... will the following work?
```
height = <&ssd1306>,"solomon,height:0";
```

I only have 128x64 displays, so I can't test the other ones :/

pelwell commented:
`height = <&ssd1306>,"solomon,height:0";` is correct. I'm not in a position to test anything beyond the overlay syntax, but for optional features like overlays best-effort is acceptable.

pelwell commented on arch/arm/boot/dts/overlays/README:
```diff
@@ -2427,6 +2427,17 @@ Params: address                 Location in display memory of first character.
         For more information refer to the device datasheet at:
         https://cdn-shop.adafruit.com/datasheets/SSD1306.pdf
 
+        
```
There's some stray whitespace on this line - please remove it.

pelwell reviewed (changes requested):
Apart from the additional parameters and the whitespace issue, you are missing an entry in the Makefile. Otherwise it looks good.

MikeDK committed:
overlays: ssd1306-spi: resolve review comments

MikeDK commented:
Thanks for the hint with the Makefile ... I always forget about that ;)

Fixed the issues, and added a parameter for the height (only height because I never saw an SSD1306 OLED with a width other than 128)

pelwell commented on arch/arm/boot/dts/overlays/ssd1306-spi-overlay.dts:
```diff
@@ -73,9 +73,10 @@
 		speed		= <&ssd1306>,"spi-max-frequency:0";
 		rotate		= <&ssd1306>,"rotate:0";
-		fps		= <&ssd1306>,"fps:0";
+		fps			= <&ssd1306>,"fps:0";
```
It looks like the stray whitespace found another home.

MikeDK replied:
gnaaa I hate tabs ;)

MikeDK committed:
overlays: ssd1306-spi: fix another stray whitespace
